### ISSUE TITLE
Amplify auth delete with added security

### ISSUE THREAD

**Issue by elusivestudio** [aws-amplify/amplify-ios#3120]
### Is your feature request related to a problem? Please describe.

I would like to request a feature for adding security layer to account deactivation. It seems when user wants to remove his account we can just call .delet() without any parameter. This is very dangerous as it can be misused if someone get access to the user's device. i.e. his child can open the app without needing to login and deactivate his account.

### Describe the solution you'd like

Add a security layer like asking the user login password and / or confirmation code that is sent to his email.

### Describe alternatives you've considered

- Very password
- Email confirmation code

### Is the feature request related to any of the existing Amplify categories?

Auth

**Comment by elusivestudio** [comment]
Yes I have to implemented a backend api to add a security layer with a confirmation code. But I thought I should report this as it would be nice if Amplify can handle it for us.

**Comment by harsh62** [comment]
This is a service dependent feature and there are currently no plans to support this in Cognito. Please open a support case with via AWS Console https://docs.aws.amazon.com/awssupport/latest/user/case-management.html to voice your opinion for a feature request in Cognito.
